File: python/scrabble-score/scrabble_score.py
def score(word):
    scores = 0
    for character in word:
        for chars, _score in {
            ("a", "e", "i", "o", "u", "l", "n", "r", "s", "t"): 1,
            ("d", "g"): 2,
            ("b", "c", "m", "p"): 3,
            ("f", "h", "v", "w", "y"): 4,
            ("k",): 5,
            ("j", "x"): 8,
            ("q", "z"): 10
        }.items():
            if character.lower() in chars:
                scores += _score
    return scores

File: go/scrabble-score/scrabble_score.go
package scrabble

import (
	"strings"
	"unicode"
)

func Score(word string) int {
	scoreMapping := map[string]int{
		"aeioulnrst": 1,
		"dg":         2,
		"bcmp":       3,
		"fhvwy":      4,
		"k":          5,
		"jx":         8,
		"qz":         10,
	}
	scores := 0

	for _, c := range word {
		for chars, score := range scoreMapping {
			index := strings.IndexRune(chars, unicode.ToLower(c))
			if index != -1 {
				scores += score
				break
			}
		}
	}
	return scores
}
